testing/test_hellsemble/hellsemble.py

```diff
@@ -291,21 +326,59 @@ def __fit_estimators_greedy(
                 fitting_history_entry = np.full((X.shape[0]), False)
                 fitting_history_entry[failed_observations_idx] = True
                 fitting_history.append(fitting_history_entry)
+
+                # add regularization - some good observations go through
+                train_score = self.evaluate_hellsemble(X_train, y_train)
+                val_score = self.evaluate_hellsemble(X_val, y_val)
+                num_additional_correct = int(
```

Comment:
can be problematic if train_score > val_score which may happen for underfitted models.